developer_tests/all_quickbuilds.sh

```diff
@@ -0,0 +1,177 @@
+#!/usr/bin/env bash
+
+# DART software - Copyright UCAR. This open source software is provided
+# by UCAR, "as is", without charge, subject to all terms of use at
+# http://www.image.ucar.edu/DAReS/DART/DART_download
+
+
+# Email build results
+TO_EMAIL_ADDRESSES='anorcio@ucar.edu'
+# Log results of all quickbuilds and compilers
+LOGFILE="all_quickbuilds_results.log"
+# Log results of failed builds only
+FAILURE_LOGFILE="all_quickbuilds_failures.log"
+
+# Store compiler arguments in an array
+FCS=( $@ )
+
+# Check if the script is running in a batch job or interactive session
+if [[ -z $PBS_ENVIRONMENT ]]; then
+  echo "ERROR: You must run this in a batch job"
+  echo "       qsub submit_me.sh"
+  echo "       or an interactive session"
+  exit 2
+fi
+
+# If no argument is provided, run with all compilers as default
+if [[ "${#FCS[@]}" == 0 ]]; then
+  FCS=( intel gnu cce nvhpc )
+fi
+
+# Include current date/time to log files
+current_date="$(printf '\n%s\n' "$(date)")"
+echo "$current_date" > "$LOGFILE"
+echo "$current_date" > "$FAILURE_LOGFILE"
+
+# Function to run quickbuild.sh for specific compilers
+function compiler_operation {
+  FC="$1"
+  # Set up DART environment variable and do a cleanup in case DART already exists
+  export DART="/glade/derecho/scratch/$USER/tmp/$FC/DART"
+  if [[ -n $FC ]]; then
+    rm -rf /glade/derecho/scratch/$USER/tmp/$FC
+  fi
+  # Create new tmp directory for a specific compiler and clone DART
+  mkdir /glade/derecho/scratch/$USER/tmp/"$FC"
+  git clone 'https://github.com/NCAR/DART.git' "$DART"
+
+  # Check if DART directory exists. If not, return an error
+  if [[ ! -d $DART ]]; then 
+    echo "No DART directory: $DART"
+    return 1
+  fi
+
+  # Specify the mkmf template for each compiler
+  if [[ $FC == "intel" ]]; then
+    template_name="mkmf.template.intel.linux"
+  elif [[ $FC == "gnu" ]]; then
```

Comment:
In addition to setting the template_name, you'll need to load the compiler module

module load gcc for gnu
module load intel for intel
module load nvhpc for nvhpc